### ISSUE TITLE
Semantic versioning for docker images.

### ISSUE THREAD

**Issue by MohamadTahir** [Infisical/infisical#485]
Hello, checking your docker hub image [Front End](https://hub.docker.com/r/infisical/frontend/tags) and [Back End](https://hub.docker.com/r/infisical/backend/tags), I noticed that they are not versioned properly, versioning the image tags properly helps with tracking changes.

**Comment by maidul98** [comment]
Hey @Grraahaam, yes this will be great. I'm just worried that when we start pushing tags with prefixes to trigger different workflows, the releases section will be altered? I'm not 100% positive this will be the case but assume the since tags will no longer solely reference the CLI, the link to the releases will be empty at times? Let me know what you think 

<img width="413" alt="Screenshot 2023-04-18 at 5 59 45 PM" src="https://user-images.githubusercontent.com/9300960/232939137-549da745-0fa7-4290-b76a-01bdcb4859db.png">

**Comment by maidul98** [comment]
Makes sense @Grraahaam and yes, it'd be nice if we only saw the change logs for the service the tag was for which should be possible since the tags will now have a prefix for the the front, back and cli. 

Also typically, the front and backend are shipped at the same time since their logic refer to each other, I'm wondering if they both should be linked to one tag vs two tags. I'm thinking it should be one tag for both, but let me know what your thoughts are as well

**Comment by maidul98** [comment]
Cool, let's leave adding the CLI within the conatiner for another thread and for this PR focus on adding individual tags

**Comment by maidul98** [comment]
Docker images now have semantic versioning
